pkg/images/defaults/useragent.go

```diff
@@ -0,0 +1,15 @@
+package defaults
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/stackrox/rox/pkg/version"
+)
+
+// UserAgent returns a default value to set the UserAgent header when sending HTTP requests to K8s API Server.
+// Use UserAgent only in central or operator applications otherwise it may panic, for more context see defaults.GetImageFlavorFromEnv
+func UserAgent() string {
+	return fmt.Sprintf("%s/v%s %s (%s/%s)", os.Args[0], version.GetMainVersion(), GetImageFlavorFromEnv().Name, runtime.GOOS, runtime.GOARCH)
```

Comment:
1. Is having command name `os.Args[0]` that useful? What would happen if we rename operator binary? I see that in PR description you mention `rhacs-operator` there but I recall that the binary is called differently in the code. Wondering how does `rhacs-operator` string actually get there.
https://github.com/stackrox/stackrox/blob/adbca26fe840695d6521f467364cb806e99a0a6b/operator/Dockerfile#L68

3. Well, instead of introducing `Name` attribute, you can split `GetImageFlavorFromEnv()` function in the part that gets and checks environment variable and use that.

```go
func UserAgent() string {
    return fmt.Sprintf("%s/v%s %s (%s/%s)", os.Args[0], version.GetMainVersion(), GetImageFlavorNameFromEnv(), runtime.GOOS, runtime.GOARCH)
}
```

```go
func GetImageFlavorNameFromEnv() (string, err) {
	envValue := strings.TrimSpace(imageFlavorEnv())
	if envValue == "" && !buildinfo.ReleaseBuild {
		envValue = ImageFlavorNameDevelopmentBuild
		log.Warnf("Environment variable %s not set, this will cause a panic in release build. Assuming this code is executed in unit test session and using '%s' as default.", imageFlavorEnvName, ImageFlavorNameDevelopmentBuild)
	}
        err := CheckImageFlavorName(envValue, buildInfo.ReleaseBuild)
        if err != nil {
          return "", err
        }
        return envValue, nil
}

// and reuse GetImageFlavorNameFromEnv in GetImageFlavorFromEnv
```

Reply:
1. Good question got this idea from the client-go default useragent code, used before that change. It also used os.Args[0] and somehow for the version this issue was observed with it used rhacs-operator. That's why assumed the example string. Do you think it would be better to use a static string set to "rhacs-operator" instead of os.Args[0]?
2. I thought about that when I was implementing but then choose to do it this way, I'll change the implementation.

Reply:
Depends on your debug configuration. When running with go run it would be the path to the temporary binary build in the background. When first building and calling executable "rhacs-operator", then running the executable and attaching debugger to the local process the string looks like in the example above.

Suggested changes are done and tested from my site.